core/src/test/java/software/amazon/awssdk/client/AsyncClientHandlerImplTest.java

```diff
@@ -152,7 +153,8 @@ private void expectRetrievalFromMocks() {
     public AsyncClientConfiguration clientConfiguration() {
         MutableClientConfiguration mutableClientConfiguration = new MutableClientConfiguration()
                 .credentialsProvider(credentialsProvider)
-                .asyncHttpClient(httpClient);
+                .asyncHttpClient(httpClient)
+                .endpoint(URI.create("http://test.com"));
```

Comment:
?

Reply:
I made the endpoint components (protocol, host) required in the HTTP request so that the interface can guarantee they're always there. The endpoint in the client configuration is the one used to determine the protocol and host that goes into the HTTP request, so they're now required here, by proxy.